src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -473,8 +473,29 @@ double Helix::safePitch()
 {
     Base::Vector3d axisVec = Axis.getValue();
     Base::Vector3d startVec = axisVec.Cross(getProfileNormal()); // pointing towards the helix start point
-    if (startVec.IsNull())
-        return Precision::Confusion(); // if the axis orthogonal to the profile, any pitch > 0 is safe
+    HelixMode mode = static_cast<HelixMode>(Mode.getValue());
+    double growthValue = Growth.getValue();
+    double turnsValue = Turns.getValue();
+
+    // handle case if axis is orthogonal to profile
+    // since startVec.IsNull() fails sometimes, for e.g. (0.0, 0.0, -0.0)
+    // we take the precision into account
+    if (startVec.Length() < Precision::Confusion()) {
+        // when not in growth mode any pitch > 0 is safe
+        if (mode != HelixMode::height_turns_growth) {
+                return Precision::Confusion();
+        }
+        // if growth is not zero, there will in many cases be intersections
+        // when the turn is >= 1, thus return an 'infinite' pitch
+        // Note: The resulting helix body is in this case often garbage since
+        // the OCC algorithm to create the helix fails.
+        // Nevertheless, the result is a valid body so it should be valuable for users
+        // to get this correct warning anyway.
+        else {
+            if (abs(turnsValue) >= 1.0 && abs(growthValue) > 0.0)
+                return INT_MAX;
```

Comment:
Wouldn't Precision::Infinite() be more appropriate? This also makes the intention clearer that "inf" should be returned.

Reply:
Additionally a function returning `double` shouldn't be returning an integer anyway (please correct me if I'm missing something here).